src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -1317,14 +1317,28 @@ def list_runtimes(cmd, os_type=None, linux=False):
     return runtime_helper.get_stack_names_only(delimiter=":")
 
 
-def list_function_app_runtimes(cmd, os_type=None):
+def list_function_app_runtimes(cmd, os_type=None, sku=None):
+    
+    from ._constants import (FLEX_RUNTIMES)
+    
     # show both linux and windows stacks by default
     linux = True
     windows = True
     if os_type == WINDOWS_OS_NAME:
         linux = False
     if os_type == LINUX_OS_NAME:
         windows = False
+    
+    is_flex = False
+    if sku is not None:
+        is_flex = sku.lower() == 'flex'
+        if not is_flex:
+            raise ValidationError("The --sku parameter only supports 'flex'.")
+        elif (is_flex and not linux):
+            raise ArgumentUsageError("The flex sku is only supported on Linux. Please try removing the --os-type parameter.")
```

Comment:
I think we wanted to ensure that info for other SKUs would still be presented.

Reply:
This is mainly to avoid confusion because we don't want to return anything if request has os-type as `windows` and sku as flex because it is not supported combination.